registry/mdns_registry.go

```diff
@@ -293,18 +359,37 @@ func (m *mdnsRegistry) Deregister(service *Service, opts ...DeregisterOption) er
 		}
 	}
 
+	// we have more than one entry remaining, we can exit
+	if len(newEntries) > 1 || newEntries[0].id == "*" {
```

Comment:
Is it possible `len(newEntries) == 0` if so then you'll get a nil pointer exception

Reply:
Good catch, i'll amend.